pkg/querier/queryrange/results_cache.go

```diff
@@ -504,6 +509,12 @@ func (s resultsCache) partition(req Request, extents []Extent) ([]Request, []Res
 		requests = append(requests, r)
 	}
 
+	// If start and end are the same (valid in promql), start == req.GetEnd() and we won't do the query.
+	// But we should only do the request if we don't have a valid cached response for it.
+	if req.GetStart() == req.GetEnd() && len(cachedResponses) == 0 {
```

Comment:
Shouldn't we just change the previous `if start < req.GetEnd() {` into `if start <= req.GetEnd() {`?

Reply:
Then we will always make an additional request if have an extent that ends at `req.GetEnd()`

Reply:
Right, got it.